bindings/rust/integration/tests/internet_http_client.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Purpose: ensure that s2n-tls is compatible with other http/TLS implementations
+///
+/// This test uses s2n-tls-hyper to make http requests over a TLS connection to
+/// a number of well known http sites.
+#[cfg(feature = "network-tests")]
+mod http_get {
+    use bytes::Bytes;
+    use http::{StatusCode, Uri};
+    use http_body_util::{BodyExt, Empty};
+    use hyper_util::{client::legacy::Client, rt::TokioExecutor};
+    use s2n_tls::{config::Config, security};
+    use s2n_tls_hyper::connector::HttpsConnector;
+    use std::{error::Error, str::FromStr};
+    use tracing_subscriber::filter::LevelFilter;
+
+    #[derive(Debug)]
+    struct TestCase {
+        pub query_target: &'static str,
+        pub expected_status_code: u16,
+    }
+
+    impl TestCase {
+        const fn new(domain: &'static str, expected_status_code: u16) -> Self {
+            TestCase {
+                query_target: domain,
+                expected_status_code,
+            }
+        }
+    }
+
+    const TEST_CASES: &[TestCase] = &[
+        // Akamai hangs indefinitely. This is also observed with curl and chrome
+        // https://github.com/aws/s2n-tls/issues/4883
+        // TestCase::new("https://www.akamai.com/", 200),
```

Comment:
Akamai does work with the current test though. It is possible it can do the TLS handshake but not an HTTP request? Do we need a separate s2n-tls-tokio-only test for some of these endpoints like you have for PQ? I'm not a huge fan of removing it for HTTP problems when the TLS part works just fine.

Reply:
That's reasonable. I changed up the `pq_client` test to a more general `tls_client` test, and added akamai there.